lib/resolvers/profile.js

```diff
@@ -22,16 +22,40 @@ module.exports = ({ models }) => ({ action, data, id }, transaction) => {
   }
 
   if (action === 'merge') {
-    const actions = [
-      Project.query(transaction).patch({ licenceHolderId: data.target }).where({ licenceHolderId: id }),
-      PIL.query(transaction).patch({ profileId: data.target }).where({ profileId: id }),
-      Permission.query(transaction).patch({ profileId: data.target }).where({ profileId: id }),
-      Role.query(transaction).patch({ profileId: data.target }).where({ profileId: id }),
-      Certificate.query(transaction).patch({ profileId: data.target }).where({ profileId: id }),
-      Exemption.query(transaction).patch({ profileId: data.target }).where({ profileId: id })
-    ];
 
-    return Promise.all(actions)
+    return Promise.resolve()
+      .then(() => Promise.all([
+        PIL.query(transaction).where({ profileId: id, status: 'active' }),
+        PIL.query(transaction).where({ profileId: data.target, status: 'active' })
+      ]))
+      .then(([pil1, pil2]) => {
+        if (pil1.length && pil2.length) {
+          throw new Error('Cannot merge profiles as both have an active PIL');
+        }
+      })
+      .then(() => {
+        return Permission.query(transaction).select().where({ profileId: id });
+      })
+      .then(permissions => {
+        const queries = permissions.map(p => {
+          return Permission.upsert({ ...p, profileId: data.target }, { establishmentId: p.establishmentId, profileId: data.target }, transaction);
```

Comment:
What's the intentional behaviour here? It looks like this will overwrite any perms on the target profile with the perms from the source profile if there's a matching establishment? Do we not want to check for the highest perm level between the two profiles, or don't we care?

Reply:
We don't care.

It's pretty trivial to manually set the permissions to be what they need to be, and in 99% of cases they'll both be `basic`